sdk/identity/identity/src/credentials/managedIdentityCredential/imdsMsi.ts

```diff
@@ -33,7 +34,14 @@ function expiresInParser(requestBody: any): number {
   }
 }
 
-function prepareRequestOptions(resource?: string, clientId?: string): PipelineRequestOptions {
+function prepareRequestOptions(
+  resource?: string,
+  clientId?: string,
+  options?: {
+    skipQuery?: boolean;
+    skipMetadataHeader?: boolean;
```

Comment:
This allows to skip both or only one of the two. From my understanding of the PR description, we are skipping both to verify the endpoint. We keep both when getting the token. What is the scenario where we would need to skip only one of the two?

Reply:
Right now we don’t have a scenario in which we might want to skip only one, but it felt worse for me to have some arbitrary parameter that would do more than one thing. Like `skipDetails`? It feels so constrained to some current vague definition. So, I went with being extra verbose on what’s the intention of this last optional parameter, in a way that can help us add more configuration later.

Reply:
Although I made this hotfix, and I’m intending to release it asap, I’m in the process of communicating this unexpected behavior to the team. It could lead to something interesting. But the code change seems an improvement regardless of future inquiries.

Reply:
Fair enough

Reply:
May be you can highlight some where in the ref-docs for the user when to use these skip-options and in what combinations?

Reply:
This is not part of the public API though. This is invisible to users. Well, except when their authentication sometimes fails (current scenario in k8s).